### ISSUE TITLE
Copy/paste

### ISSUE THREAD

**Issue by edemaine** [edemaine/cocreate#94]
This would be particularly useful for pasting across boards or pages.

**Comment by jkitchin** [comment]
I would like to paste screenshots into cocreate so that I can annotate them.

**Comment by edemaine** [comment]
@jkitchin Yep, that is issue #22.

**Comment by duetosymmetry** [comment]
Sorry, I commented in the wrong place (I had too many tabs open!). The following comment was intended for this issue:
https://github.com/edemaine/cocreate/issues/17#issuecomment-705972664

**Comment by edemaine** [comment]
Yeah, I have some experience in [Coauthor](https://github.com/edemaine/coauthor/) with defining a custom data type for internal representations of data. Oh, actually, looks like I only [did so for dragging](https://github.com/edemaine/coauthor/blob/1f4d02d156853fb3ff80619effb48fa542c1ee6f/client/message.coffee#L791-L793) (also [here](https://github.com/edemaine/coauthor/blob/1f4d02d156853fb3ff80619effb48fa542c1ee6f/client/message.coffee#L532-L534)), but I assume/hope the clipboard can behave similarly. This would let Cocreate -> Cocreate copy/paste work with the internal database representations.

Although the example app you linked doesn't fill a `image/svg+xml` slot, I imagine it *could*, for apps that support it. Worth testing with programs like Inkscape and Illustrator.

**Comment by edemaine** [comment]
For those following this issue, cut/copy/paste via Ctrl/Command-X/C/V now works quite well. In particular, you can now copy objects from one page to another, or even between boards!  You can also paste links to images and they'll get embedded (part of #144), or paste text and it'll turn into a text object (part of #142).

**Comment by ashawdr** [comment]
@edemaine I'm not quite sure what I might be doing wrong, but I cannot seem to copy text from one board to the next using crtrl C then V. I can't even copy/paste text within the same board. Any advice?

**Comment by edemaine** [comment]
What server, operating system, and browser are you using?  For CSAIL, Windows, Chrome it works well.

**Comment by ashawdr** [comment]
Gather.Town, Windows, and Chrome. I think the issue might be with Gather, I'm just not quite sure how to fix it.

**Comment by edemaine** [comment]
Yeah, Gather.Town is running an older version of Cocreate, which doesn't have this feature. We need to wait until they update...

**Comment by edemaine** [comment]
I think I'm now convinced that Ctrl-V should paste in the center of the view, or mouse cursor if it's been seen since the last `pointerleave` event. (This matches Inkscape behavior.)

Ctrl-Alt-V (Inkscape's shortcut) or Ctrl-Shift-V could be the new way to access "paste in place". It's useful e.g. when copying things into previously duplicated pages.
